File: backend/app/models/participant.py
from datetime import datetime, timezone
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base


class Participant(Base):
    __tablename__ = "participants"

    id = Column(Integer, primary_key=True, index=True)
    meeting_id = Column(Integer, ForeignKey("meetings.id"), nullable=False)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=True)  # nullable for guest users
    display_name = Column(String, nullable=False)
    role = Column(String, nullable=False, default="participant")  # "host" or "participant"
    is_muted = Column(Integer, default=0)
    joined_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    left_at = Column(DateTime, nullable=True)

    # Relationships
    meeting = relationship("Meeting", back_populates="participants")
    user = relationship("User", back_populates="participations")

File: frontend/types/participant.ts
import { Meeting } from "./meeting";

export interface Participant {
  id: number;
  display_name: string;
  role: "host" | "participant";
  is_muted: number;
  joined_at: string;
  left_at: string | null;
  meeting_id: number;
}

export interface JoinMeetingRequest {
  display_name: string;
}

export interface JoinMeetingResponse {
  meeting: Meeting;
  participant: Participant;
}
